pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -363,8 +363,15 @@ pub mod pallet {
 
 			let currency_length = currencies.len();
 
-			let currency_ids: BoundedVec<FungiblesAssetIdOf<T>, T::MaxCurrenciesPerPool> =
-				T::NextAssetIds::try_get(currency_length.saturated_into()).map_err(|err| err.into())?;
+			let currency_ids = T::NextAssetIds::try_get(currency_length.saturated_into())
+				.map_err(|_| Error::<T>::Internal)
+				.and_then(|ids| {
+					if ids.len() != currency_length {
+						log::error!(target: LOG_TARGET, "NextAssetIds::try_get returned wrong number of ids");
+						return Err(Error::<T>::Internal);
+					}
```

Comment:
this, on the other hand, makes sense, because it indicates inactionable misconfiguration.